cli/azd/pkg/tools/github/github_test.go

```diff
@@ -0,0 +1,73 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/test/mocks"
+	"github.com/azure/azure-dev/cli/azd/test/mocks/mockinput"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewGitHubCli(t *testing.T) {
+	configRoot := t.TempDir()
+	t.Setenv("AZD_CONFIG_DIR", configRoot)
+
+	mockContext := mocks.NewMockContext(context.Background())
+
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet && request.URL.Host == "github.com"
+	}).Respond(&http.Response{
+		StatusCode: http.StatusOK,
+		Body:       io.NopCloser(bytes.NewBufferString("this is github cli")),
+	})
+
+	mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
+		return strings.Contains(args.Cmd, "gh") && len(args.Args) == 1 && args.Args[0] == "--version"
+	}).Respond(exec.NewRunResult(
+		0,
+		fmt.Sprintf("gh version %s (abcdef0123)", cGitHubCliVersion.String()),
+		"",
+	))
+
+	mockExtract := func(src, dst string) (string, error) {
+		exp, _ := azdGithubCliPath()
+		_ = os.Rename(src, exp)
+		return src, nil
+	}
```

Comment:
@vhvb1989 The extraction logic is probably where I care the most about currently as it is non-trivial. Can we just embed `tar.gz` and `.zip` containing a dummy `gh` file? We can continue mocking out the version call.

Reply:
added tests for extract